src/utils/url.js

```diff
@@ -15,6 +22,47 @@ export const getFullUrl = (url: string, realm: string): string =>
 export const isUrlOnRealm = (url: string, realm: string): boolean =>
   url.startsWith('/') || url.startsWith(realm);
 
+export const isUrlInAppLink = (url: string, realm: string) : boolean =>
+  (isUrlOnRealm(url, realm) ? url.split(realm).pop().startsWith('/#narrow') :
+  false);
+
+export const isMessageLink = (url: string, realm: string) : boolean =>
```

Comment:
Noticing few super-minor differences in formatting (should be no space before `:`.
Install Prettier.js in your IDE and try using it. Maybe ping me in chat to see how to configure it best.